openqa-investigate

```diff
@@ -119,6 +129,37 @@ trigger_jobs() {
     fi
 }
 
+check_dependencies_to_determine_whether_to_skip_or_postpone() {
+    local id=$1 job_data=$2 parallel_parents dependency_data pending_cluster_jobs cluster_jobs relevant_cluster_results
+
+    # skip parallel children (will be handled together with parallel parent)
+    # note: For this to work, this script needs to be executed for all job results (so we won't miss the investigation in case the parallel parent has passed).
+    parallel_parents=$(echo "$job_data" | runjq -r '(.job.parents["Parallel"] | length)') || return $?
+    [[ $parallel_parents != 0 ]] \
+        && echoerr "Skipping investigation of job $id: Investigation will be done on the parallel parent" && return 0
+
+    # postpone if not all dependencies are done/cancelled
+    # note: This "AJAX" route is normally used to render the dependencies tab in the web UI.
+    dependency_data=$(openqa-cli "${client_args[@]}" --apibase '' --json tests/"$id"/dependencies_ajax)
+    pending_cluster_jobs=$(echo "$dependency_data" | runjq -r '[.nodes[] | select(.state != "done" and .state != "cancelled")] | length') || return $?
+    [[ $pending_cluster_jobs != 0 ]] \
+        && echoerr "Postponing to investigate job $id: waiting until pending dependencies have finished" && return 142
+
+    # determine number of jobs in cluster with relevant result
+    # note: The "$id, …" is to always also consider the job itself (even if it is just a single job and therefore not in a cluster).
+    cluster_jobs=$(echo "$dependency_data" | runjq -r "[$id, [.cluster[] | select(contains([$id]))]] | flatten | unique") || return $?
+    # shellcheck disable=SC2016
+    relevant_cluster_results=$(echo "$dependency_data" | runjq --argjson cluster_jobs "$cluster_jobs" -r '[.nodes[] | select([.id] | inside($cluster_jobs)) | .result | select(inside("failed", "incomplete", "timeout_exceeded"))] | length') || return $?
```

Comment:
Looks like this is actually wrong and should be

```suggestion
    relevant_cluster_results=$(echo "$dependency_data" | runjq --argjson cluster_jobs "$cluster_jobs" -r '[.nodes[] | select([.id] | inside($cluster_jobs)) | .result | select(inside("failed"))] | length') || return $?
```
because for incomplete/timeout_exceeded we only run `openqa-label-known-issues-hook`.

Reply:
correct